diff --git a/darkflow/net/flow.py b/darkflow/net/flow.py
index a5b8ceb..bb60704 100644
--- a/darkflow/net/flow.py
+++ b/darkflow/net/flow.py
@@ -90,7 +90,8 @@ def return_predict(self, im):
                 "y": tmpBox[2]},
             "bottomright": {
                 "x": tmpBox[1],
-                "y": tmpBox[3]}
+                "y": tmpBox[3]},
+            "coloridx": tmpBox[5]
         })
     return boxesInfo
 
@@ -142,4 +143,4 @@ def predict(self):
 
         # Timing
         self.say('Total time = {}s / {} inps = {} ips'.format(
-            last, len(inp_feed), len(inp_feed) / last))
\ No newline at end of file
+            last, len(inp_feed), len(inp_feed) / last))
